File: src/main/java/Logger.java
interface ILogger {
    void Log(String message);
}

class FileLogger implements ILogger {
    @Override
    public void Log(String message) {
        // write to the file
    }
}

class ConsoleLogger implements ILogger {
    @Override
    public void Log(String message) {
        // print to console
    }
}

class AggregateLogger {
    //ToDo: Implement Aggregate logger.
}

File: src/main/kotlin/Logger.kt
internal interface IKotlinLogger {
    fun log(message: String?)
}

internal class KotlinFileLogger : IKotlinLogger {
    override fun log(message: String?) {
        println("File logger call. Message  => $message")
    }
}

internal class KotlinConsoleLogger : IKotlinLogger {
    override fun log(message: String?) {
        println("Console logger call. Message => $message")
    }
}

//ToDo: Implement Aggregate logger.
internal class KotlinAggregateLogger